src/main/java/org/victorrobotics/frc/dtlib/actuator/motor/ctre/DTTalonFX.java

```diff
@@ -0,0 +1,297 @@
+package org.victorrobotics.frc.dtlib.actuator.motor.ctre;
+
+import org.victorrobotics.frc.dtlib.actuator.motor.DTMotor;
+import org.victorrobotics.frc.dtlib.actuator.motor.DTMotorFaults;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+import com.ctre.phoenix.motorcontrol.Faults;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.TalonFXControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+public class DTTalonFX implements DTMotor<WPI_TalonFX, SupplyCurrentLimitConfiguration> {
+    private static final double TICKS_PER_REV      = 2048;
+    private static final double SECONDS_PER_MINUTE = 60;
+
+    private final WPI_TalonFX internal;
+    private final int         deviceID;
+
+    private String firmwareVersion;
+
+    private double pidP;
+    private double pidI;
+    private double pidD;
+    private double pidF;
+    private double pidIZ;
+
+    private boolean brakeEnabled;
+
+    public DTTalonFX(int canID) {
+        this(canID, "");
+    }
+
+    public DTTalonFX(int canID, String canBus) {
+        internal = new WPI_TalonFX(canID, canBus);
+        deviceID = canID;
+    }
+
+    @Override
+    public WPI_TalonFX internal() {
```

Comment:
You might consider a different name for the motor object and its accessor method.

Reply:
I looked at several options, but this one made the most sense to me. If you have other suggestions, I'd love to hear them.

Reply:
A couple ideas: `getMotor()` `getMotorImpl()`. Or leave as is.

Reply:
I don’t like `DTMotor::getMotor()`, that just sounds like a redundant method that does nothing. `DTMotor::getMotorImpl()` is a decent option, I’ll switch it to that. I was looking for a name that would scare away beginner programmers (reduce complexity) and invite curious intermediates to inspect it.